content/docs/integrating-qri/r-studio.md

```diff
@@ -0,0 +1,201 @@
+# it does the work to read the schema from the qri dataset so we can
+# explicitly set col_types
+getQriDataset <- function(datasetReference) {
+
+  # first, get the schema (schema is a child of the qri structure component)
+  # parse the json, schema will be a nested list
+  structure <- system2(
+    'qri',
+    args = str_interp('get structure ${datasetReference} --format json'),
+    stdout = TRUE
+  ) %>% parse_json()
+
+  # next, map the jsonSchema types into col_types
+  # types is a string of letter abbreviations for column types
+  # (character = c, integer = i, etc) for example, if the dataset had 2 string
+  # columns, 3 number columns, and 1 integer column, types would be "ccdddi"
+  types <- paste(
+    map_chr(
+      structure$schema$items$items,
+      function(d) { return(mapType(d$type)) }
+    ),
+    collapse = ''
+  )
+
+  # now that we have a string of column types, we can use read_csv() to import
+  # the dataframe to actually get the CSV string, we use system2() to call
+  # qri get ... bear in mind that qri doesn't enforce column types unless you
+  # ask it to, so it's possible that the values in the dataset's body may not
+  #match the dataset's schema
+  df = read_csv(
+    system2(
+      'qri',
+      args=str_interp('get body ${datasetReference}'),
+      stdout = TRUE
+    ),
+    col_names = TRUE,
+    col_types = types
+  )
+  return(df)
+}
+```
+
+With `getQriDataset()` defined, let's use it to import the turnstile data.
+
+```{r}
+turnstile_counts_2020 <- getQriDataset('nyc-transit-data/turnstile_daily_counts_2020')
+
+glimpse(turnstile_counts_2020)
+
+Rows: 113,033
+Columns: 12
+$ stop_name      <chr> "Astoria - Ditmars Blvd", "Astoria - Ditmars Blvd", "Astoria - Ditmars Blvd", …
+$ daytime_routes <chr> "N W", "N W", "N W", "N W", "N W", "N W", "N W", "N W", "N W", "N W", "N W", "…
+$ division       <chr> "BMT", "BMT", "BMT", "BMT", "BMT", "BMT", "BMT", "BMT", "BMT", "BMT", "BMT", "…
+$ line           <chr> "Astoria", "Astoria", "Astoria", "Astoria", "Astoria", "Astoria", "Astoria", "…
+$ borough        <chr> "Q", "Q", "Q", "Q", "Q", "Q", "Q", "Q", "Q", "Q", "Q", "Q", "Q", "Q", "Q", "Q"…
+$ structure      <chr> "Elevated", "Elevated", "Elevated", "Elevated", "Elevated", "Elevated", "Eleva…
+$ gtfs_longitude <dbl> -73.91203, -73.91203, -73.91203, -73.91203, -73.91203, -73.91203, -73.91203, -…
+$ gtfs_latitude  <dbl> 40.77504, 40.77504, 40.77504, 40.77504, 40.77504, 40.77504, 40.77504, 40.77504…
+$ complex_id     <chr> "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1"…
+$ date           <chr> "2020-01-01", "2020-01-02", "2020-01-03", "2020-01-04", "2020-01-05", "2020-01…
+$ entries        <int> 5812, 15506, 15959, 9265, 7191, 16810, 17436, 17575, 17554, 17529, 10304, 7418…
+$ exits          <int> 6659, 14307, 14500, 9161, 7743, 14919, 14765, 15340, 15112, 15250, 9816, 8157,…
+```
+
+There it is, and all the columns got their types from the qri dataset!
+
+## Get your R on
```

Comment:
This is the corporate thought police. Here to water down all language to soupy mush:
```suggestion
## Explore!
```